### PR TITLE
Allow x86 with x86_DBXUpdate

### PR DESCRIPTION
**Pull request by ericbl**
add simple flag to use the x86_DBXUpdate_2025-10-14.bin for x86 platform.
Tested on a Lenovo Thinkpad 8 tablet, with IA32 / x86 EFI

### CODE REVIEW

**Comment by cjee21** [comment]
Sure this works? Does it correctly select x86 if CPU is x64 capable but UEFI is x86?

File: README.md [hunk 1]
```diff
@@ -3,7 +3,7 @@
...
 
 > [!IMPORTANT]
-> The DBX checking in `Check UEFI PK, KEK, DB and DBX` is made for x64 and arm64 systems. If you are using an x86 or arm system, it is necessary to replace the `*.bin` files with ones for your system architecture and edit their filenames in the PowerShell script (`Check UEFI PK, KEK, DB and DBX.ps1`) accordingly. The `*.bin` files for various architectures can be obtained from [github.com/microsoft/secureboot_objects](https://github.com/microsoft/secureboot_objects/tree/main/PostSignedObjects/DBX).
+> The DBX checking in `Check UEFI PK, KEK, DB and DBX` is made primarly for x64 and arm64 systems and later extended for x86. As done for x86, if you are using an IA32 arm system, it is necessary to replace the `*.bin` files with ones for your system architecture and edit their filenames in the PowerShell script (`Check UEFI PK, KEK, DB and DBX.ps1`) accordingly. The `*.bin` files for various architectures can be obtained from [github.com/microsoft/secureboot_objects](https://github.com/microsoft/secureboot_objects/tree/main/PostSignedObjects/DBX).
```
Comment (cjee21) [inline README.md:6]: No need `and later extended`. The ARM64 was 'later extended' as well.

`IA32 arm` does not make sense. There is no such thing. 32-bit arm is officially just `arm`.

**Review by cjee21 — CHANGES_REQUESTED** [review]
Waiting for answer to above question and changes to readme.

**Comment by ericbl** [comment]
> Sure this works? Does it correctly select x86 if CPU is x64 capable but UEFI is x86?

sorry, actually, you are right: the script is working just with the x64 part, so my PR is currently useless in my system: 
The `(Get-WmiObject Win32_Processor -ErrorAction Stop).Architecture` is returning 9 despite running a Windows 10 32 bits with EFI32 bits. But the CPU Atom Z3770 supports x64.
`wmic os get osarchitecture` returns 32 bits. 
Trying with a better check,

**Comment by cjee21** [comment]
This is why I have not added 32-bit x68 to script till now. I do not have a system to test and I do not know of a sure way to determine UEFI bit-ness without testing.

**Comment by cjee21** [comment]
Looks like Windows bit-ness must match UEFI bit-ness? If that is true we should be checking Windows bit-ness. But seems there may be edge cases of people doing workarounds to install Windows that does not match UEFI bit-ness.

**Comment by cjee21** [comment]
@ericbl I tried to add support for all architectures. Please test the latest main and see if it works.

**Comment by ericbl** [comment]
I did similar code change on my side and pushed on same time as your update on main, by checking Windows `[Environment]::Is64BitOperatingSystem`  too.
It is running properly the x86 bin on my tablet.

But I am not sure if it is was we want or need.

Windows 32 bits (up to win10 since no 32bits on win11) can be installed on any x64 machine ~~with x64 EFI~~. 
On x64 system, either Windows 32 bits or Windows x64 can be installed.

Only very few systems have a 32 bits EFI (IA32 EFI), like my tablet from 2014 with Atom CPU with 2 GB of RAM.
On IA32 EFI, only Windows 32 bits can be installed. Most Linux have also issue to support IA32 EFI.

There is much more PC in the 2010-2015 age that we sold with Windows 32 bits on EFI x64 than those on IA32 EFI.

I suppose we want to check only the EFI bit-ness, but I haven't find yet an easy way to check this, aside mounting the EFI partition (with `mountvol /s`) and checking the EFI/Boot folder:
it is  bootx64.efi for x64 EFI (like on my PC) while it is bootia32.efi on the tablet.

**Comment by cjee21** [comment]
> On x64 EFI, either Windows 32 bits or Windows x64 can be installed.

Hmm...

> - UEFI doesn't support cross-platform boot.
>   - UEFI x64 devices can only run a 64-bit operating system. Most modern UEFI devices are x64.
>   - UEFI x86 devices can run only a 32-bit operating system. For Windows, only Windows 10 x86 supports UEFI x86. Windows 11 and newer doesn't support UEFI x86. Lack of UEFI x86 support in Windows 11 generally isn't an issue since UEFI x86 devices are rare.

Source: https://learn.microsoft.com/en-us/windows/deployment/windows-deployment-scenarios-and-tools

**Comment by ericbl** [comment]
Current output of the `Check UEFI PK, KEK, DB and DBX.cmd` on my Lenovo Thinkpad 8 tablet with EFI IA32

On previous main, using x64 db:


```
Checking for Administrator permission...
Running as administrator - continuing execution...

11 Februar 2026
Manufacturer: LENOVO
Model: 20BN000UGE
BIOS: LENOVO, GTET79WW (1.79), GTET79WW (1.79), LENOVO - 1790
Windows version: 22H2 (Build 19045.6809)

Secure Boot status: Enabled

Current UEFI PK
√ Lenovo Ltd. PK CA 2012

Default UEFI PK
√ Lenovo Ltd. PK CA 2012

Current UEFI KEK
√ Microsoft Corporation KEK CA 2011 (revoked: False)
√ Microsoft Corporation KEK 2K CA 2023 (revoked: False)
√ Lenovo Ltd. KEK CA 2012 (revoked: False)

Default UEFI KEK
√ Microsoft Corporation KEK CA 2011 (revoked: False)
X Microsoft Corporation KEK 2K CA 2023                                                                                  
√ Lenovo Ltd. KEK CA 2012 (revoked: False)                                                                         

Current UEFI DB
√ Microsoft Windows Production PCA 2011 (revoked: False)
√ Microsoft Corporation UEFI CA 2011 (revoked: False)
√ Windows UEFI CA 2023 (revoked: False)
√ Microsoft UEFI CA 2023 (revoked: False)
√ Microsoft Option ROM UEFI CA 2023 (revoked: False)
√ ThinkPad Product CA 2012 (revoked: False)

Default UEFI DB
√ Microsoft Windows Production PCA 2011 (revoked: False)
√ Microsoft Corporation UEFI CA 2011 (revoked: False)
X Windows UEFI CA 2023
X Microsoft UEFI CA 2023
X Microsoft Option ROM UEFI CA 2023
√ ThinkPad Product CA 2012 (revoked: False)
√ Lenovo UEFI CA 2014 (revoked: False)

Current UEFI DBX
2025-10-14 (v1.6.0) : FAIL: 404 failures, 27 successes detected
Windows Bootmgr SVN : None
Windows cdboot SVN  : None
Windows wdsmgfw SVN : None

Press any key to continue . . .
```

On my current branch (and should be similar on current main):
same result but only the last part
`Current UEFI DBX
2025-10-14 (v1.6.0) for x86 : FAIL: 4 failures, 40 successes detected`

So for my tablet, I am not even sure if the x86 bin is necessary since the one for x64 works.
But I am not confident yet to run the `Apply 2023 KEK, DB and bootmgfw update`.

**Comment by cjee21** [comment]
> should be similar on current main

Please test to be sure in case there are bugs in my code.

> I am not even sure if the x86 bin is necessary

Yes it is. DBX is different for every arch since UEFI can only boot matching binaries. x64 DBX only contains revocations for x64 efi files and so on. As you can see, using correct DBX result in more success and less fail which is the correct result. You are just missing 4 revocations. You do not need to revoke the 404 x64 binaries on a x86 system.

**Comment by ericbl** [comment]
> > On x64 EFI, either Windows 32 bits or Windows x64 can be installed.

> Source: https://learn.microsoft.com/en-us/windows/deployment/windows-deployment-scenarios-and-tools

ok, my bad. So all older systems with x64 CPU running Windows 32 bits were never using EFI, but only the legacy mode.
Are we sure your script CANNOT run when we have legacy system?
Maybe the script should check
`$env:firmware_type`
 must be `UEFI`, and the script should be skipped for `legacy` systems.

**Comment by cjee21** [comment]
> Are we sure your script CANNOT run when we have legacy system?

It will run but will show secure boot status "Not available" and quit immediately.

Your tablet likely has x64 CPU with x86 UEFI and thus x86 Windows and it will not be able to run x64 Windows.

> I am not confident yet to run the Apply 2023 KEK, DB and bootmgfw update

You do not have to if Windows auto applies it for you. Windows will auto apply it in the coming months if it has high confidence that applying it will not cause issues on your system. Some systems have corrupt UEFI after applying the update due to firmware bugs. But if your tablet is too rare, I do not know if Microsoft will have enough diagnostic data to have high confidence for it.

**Comment by cjee21** [comment]
I looked again at your output carefully, looks like Windows already applied the update for you. You can check with the Check Windows state script. So what you need to run is not Apply 2023 KEK, DB and bootmgfw update but the Apply DBX update one.

**Comment by ericbl** [comment]
your main branch is working properly on my IA32 tablet, giving the same output as with my script, after windows update:

```
Windows version: 22H2 (Build 19045.6937)
Detected x86 UEFI architecture. Ensure that this is correct for valid DBX results.
(...)
2025-10-14 (v1.6.0) : FAIL: 4 failures, 40 successes detected
```

and yes, I also see the DB update was done already. 
Today's Windows update (Feb 2026) is likely to push the certificate updates on more systems..
As for DBX update, not sure it would ever come on my systems. And the `high confidence` param from MS is not very reliable when you look at some comments on recent [MS Ask Me Anything session regarding SecureBoot certificates](https://techcommunity.microsoft.com/event/windowsevents/ask-microsoft-anything-secure-boot/4486023). The KEK update last month completely crashed/froze my desktop PC from 2014 (which may not support KEK update).

Back to your repo, thanks for the updates of the day. I just have 2 more suggestions regarding the documentation:

`2025-10-14 (v1.6.0) : FAIL: 4 failures, 40 successes detected`
I added on this branch the
`2025-10-14 (v1.6.0) **for x86** : FAIL: 4 failures, 40 successes detected`
to clearly output the bin used. 

even if you added above
`Detected x86 UEFI architecture. Ensure that this is correct for valid DBX results.`

In the README,
` The script attampts to detect the UEFI architecture by assuming the installed Windows architecture matches`
attempts for the spelling error.
but I believe your should reformulate this by linking the doc from MS as you did above: as from the spec, Windows archi == EFI archi.
so could be something like:
` The script is detecting the Windows and CPU architectures (32 vs 64 bits, ARM vs x86/x64). Since [UEFI doesn't support cross-platform boot ](https://learn.microsoft.com/en-us/windows/deployment/windows-deployment-scenarios-and-tools), the Windows architecture is the UEFI architecture.`

IA32 EFI are already rare. ARM 32 bits EFI supporting Windows 10 even more. AFAIK, only Windows RT (8.1) was on ARM32. Aside Windows 10 Mobile, I tough no Windows 10 ARM officially existed, but apparently, [some guys managed to install some version of Windows 10 ARM32  on their SurfaceRT](https://xdaforums.com/t/tutorial-install-windows-10-for-arm32-on-your-surface-rt.4052035/)

**Comment by cjee21** [comment]
> As for DBX update, not sure it would ever come on my systems

Use the Apply DBX update script to trigger Windows to re-apply the DBX updates. See if that makes it green.

> the Windows architecture is the UEFI architecture

This only true for officially supported systems. Some people may use Linux loader to load a Windows loader of different architecture than the UEFI architecture.

Thanks for all the help and suggestions. I will incorporate them shortly.
